monitoring/bin/create_elasticsearch_datasource_cluster.sh

```diff
@@ -16,50 +16,55 @@ export LOG_NS="${LOG_NS:-logging}"
-  log_error "No monitoring components found in the $MON_NS namespace.  Monitoring needs to be deployed in this namespace in order to configure Elasticsearch with this script.";
+  log_error "No monitoring components found in the $MON_NS namespace."
+  log_error "Monitoring needs to be deployed in this namespace in order to configure Elasticsearch with this script.";
   exit 1
 else
-  log_verbose "Monitoring found in $MON_NS namespace.  Continuing."
+  log_debug "Monitoring found in $MON_NS namespace.  Continuing."
 fi
 
 # Check to see if logging namespace provided exists and components have already been deployed
 log_info "Checking for Elasticsearch pods in the $LOG_NS namespace ..."
 if [[ $(kubectl get pods -A -l app=v4m-es -o custom-columns=:metadata.namespace --no-headers | uniq | wc -l) == 0 ]]; then
-  log_error "No logging components found in the $LOG_NS namespace.  Logging needs to be deployed in this namespace in order to configure Elasticsearch with this script.";
+  log_error "No logging components found in the $LOG_NS namespace."
+  log_error "Logging needs to be deployed in this namespace in order to configure Elasticsearch with this script.";
 else
-  log_verbose "Logging found in $LOG_NS namespace.  Continuing."
+  log_debug "Logging found in $LOG_NS namespace.  Continuing."
 fi
 
 # Create temporary directory for string replacement in the grafana-datasource-es.yaml file
   monDir=$TMP_DIR/$MON_NS
   mkdir -p $monDir
-  cp -R monitoring/grafana-datasource-es.yaml $monDir/grafana-datasource-es.yaml
+  cp monitoring/grafana-datasource-es.yaml $monDir/grafana-datasource-es.yaml
 
-# Pull Elasticsearch password in temp directory file
+# Retrieving Elasticsearch password for data source setup
 adminPass="$(kubectl -n $LOG_NS get secret internal-user-admin -o=jsonpath="{.data.password}" | base64 --decode)"
 
 # Replace placeholders
-log_debug "Replacing logging namespace for files in [$monDir]"
+log_debug "Replacing variables in $monDir/grafana-datasource-es.yaml file"
 if echo "$OSTYPE" | grep 'darwin' > /dev/null 2>&1; then
   sed -i '' "s/__sourceName__/Elastic/g" $monDir/grafana-datasource-es.yaml
   sed -i '' "s/__namespace__/$LOG_NS/g" $monDir/grafana-datasource-es.yaml
   sed -i '' "s/__userID__/admin/g" $monDir/grafana-datasource-es.yaml
   sed -i '' "s/__passwd__/$adminPass/g" $monDir/grafana-datasource-es.yaml
 else
   sed -i "s/__sourceName__/Elastic/g" $monDir/grafana-datasource-es.yaml
-  sed -i "s/__namespace__/$logNS/g" $monDir/grafana-datasource-es.yaml
+  sed -i "s/__namespace__/$LOG_NS/g" $monDir/grafana-datasource-es.yaml
   sed -i "s/__userID__/admin/g" $monDir/grafana-datasource-es.yaml
   sed -i "s/__passwd__/$adminPass/g" $monDir/grafana-datasource-es.yaml
 fi
-  
+
+if [ ! -z "$(kubectl get secret -n $MON_NS grafana-datasource-es -o custom-columns=:metadata.name --no-headers --ignore-not-found)"]; then
```

Comment:
Mostly a "style" thing but you could replace `! -z` with `-n`...but we haven't adopted formal coding conventions or standardization.